### PR TITLE
feat: multi-crate

### PR DESCRIPTION
**Pull request by svmnotn**
The name of the folder could be anything.

### CODE REVIEW

File: graphql-parser/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+name = "graphql-parser"
+version = "0.1.0"
+authors = ["cscks <cs.cksac@gmail.com>"]
+license = "MIT"
+readme = "README.md"
+repository = "https://github.com/cksac/graphql-rs"
+homepage = "https://github.com/cksac/graphql-rs"
+
+[lib]
+name = "graphql_parser"
```
Comment (cksac) [inline graphql-parser/Cargo.toml]: I would prefer graphql-parser show in the crates.io

**Comment by cksac** [comment]
thanks, I would like the crate name to use "-" instead of "_", to keep consistent with graphql-rs

**Comment by svmnotn** [comment]
The name of the crate as seen in crates.io will be with a `-` due to it being obtained from the `[packge]` section however rust does not allow such names to exist as crates within a `extern crate` definition.

So, when using the library, it will be called as `extern crate graphql_rs` not `extern crate graphql-rs`

**Comment by calebmer** [comment]
@cksac I work with @svmnotn and we are assigned to code review each other, so I'll also be doing code reviews if you don't mind :blush:

I think the goal here with the multi crate idea is to get to a point which is like [Diesel](https://github.com/sgrif/diesel) where our crates are top level directories, so:

```
+---graphql
|   |---src
|   |---Cargo.toml
+---graphql-parser
|   |---src
|   |---Cargo.toml
```

Where both `graphql` and `graphql-parser` are published to Cargo. As a side note, I think a long term goal should be acquiring the `graphql` crate name. Currently the person using it has [this](https://github.com/countcain/graphql-rust) empty repository. Once the project is further along we should try to get that name.

**Comment by cksac** [comment]
@calebmer peer review will be great. For the crate name, I am not sure is it transferable. If ok, it would be great to get that name when this project made some progress.

**Comment by calebmer** [comment]
Even if the name isn't transferable, we could publish on top of what's already there.

**Pull request merged**